tech-docs/DRIFTDETECTION.md

```diff
@@ -0,0 +1,46 @@
+# Drift Detection
+
+## Overview
+
+In general, Terraform enables you to provision and manage your Infrastructure as Code. The management part also comprises the ability to detect and reconcile configuration drifts in your infrastructure.
+Basically this is achieved by executing the `terraform plan` with the option `-detailed-exitcode`. This will return an exit code of `2` if there are changes to be applied, `1` if there is an error, and `0` if the infrastructure matches the Terraform state.
```

Comment:
I'm wondering if we should mention that this relies on Terraform detecting these state deviations via implicitly refreshing the state on a plan (which could also be executed alone with the `--refresh-only`) flag.  and then acts based on the updated state. 

Then Terraform is able to detect deviations to the as is state. Running `-detailed-exitcode` then as usual returns `2` in case of detected required actions to get from the as is state to the to be state. When not having done any other changes, this can be seen as an indication for a configuration drift.

(I'd expect given e.g. I change something and in the background there's a configuration drift, then I'd also get `2`, or if I just change something without a configuration drift in the sense of manual changes in the background)

Reply:
It makes sense to add a sentence on the mechanics (or referencing the Terraform documentation there if I find a fitting piece).

The `refresh-only` flag is from my understanding already a first step in dealing with the State drift i.e., transferring the changes on the platform into the state. 

I think it makes sense to add an additional section on how to deal with the state drift that was detected referencing the official Terraform documentation where applicable as next steps after detecting the drift.

Reply:
Updated the sections